docs/usage/guides/cuda/build.sh

```diff
@@ -1,15 +1,27 @@
 #!/bin/bash
-set -e
-cd $(dirname $0)
 
-K3S_TAG="${1:-v1.18.10+k3s1}"
-IMAGE_TAG="${K3S_TAG/+/-}"
+export CI_REGISTRY_IMAGE="YOUR_REGISTRY_IMAGE_URL"
+export VERSION="1.0"
+export K3S_TAG="v1.21.2+k3s1"
+export DOCKER_VERSION="20.10.7"
+export IMAGE_TAG="v1.21.2-k3s1"
 
-if [ -d k3s ]; then
-    rm -rf k3s
+sudo docker build -f Dockerfile.base --build-arg DOCKER_VERSION=$DOCKER_VERSION -t $CI_REGISTRY_IMAGE/base:$VERSION . && \
+sudo docker push $CI_REGISTRY_IMAGE/base:$VERSION
+
+sudo rm -rf ./k3s && \
+git clone --depth 1 https://github.com/rancher/k3s.git -b "$K3S_TAG" && \
+sudo docker run -ti -v ${PWD}/k3s:/k3s -v /var/run/docker.sock:/var/run/docker.sock $CI_REGISTRY_IMAGE/base:1.0 sh -c "cd /k3s && make" && \
+sudo ls -al k3s/build/out/data.tar.zst
```

Comment:
Do we actually need a custom image here?
I guess something like `docker run --rm -v ${PWD}/k3s:/k3s -v /var/run/docker.sock:/var/run/docker.sock -w /k3s docker:20.10.7 make` would already work, right?
If I understand the k3s build process correctly, everything will run with dapper anyway :thinking:

Reply:
I don't think you have necessary build tools in the `docker:20.10.7` container, that's why I've building the base build image first with `build-essential` on board

Reply:
I've removed the sudo commands from the `build.sh` but I've initially added them so the user 100% won't forget to run it with sudo. I think that the build actually creates `k3s/build/out/data.tar.zst` file with root permissions because of downstream use of root so you need to use sudo to be able to use it after that.

Reply:
I actually just gave it a try and it works (but needs an `apk add make`, so it would be `sh -c "apk add make && make"`) :+1: 
All the rest runs inside dapper, so no extra tools requires if I got it right :thinking: 
But also not a huge blocker.

Reply:
Well one extra reason not to use Alpine based image for me was to avoid any possible extra issues between Alpine/Ubuntu libraries & etc, that's why I'm using `nvidia/cuda:11.2.0-base-ubuntu18.04` image for both build & runtime images.